src/Container.php

```diff
@@ -665,6 +665,34 @@ public function share($key, $value, $protected = false)
         return $this->set($key, $value, true, $protected);
     }
 
+    /**
+     * Create a lazy proxy resource for given class, and register it in the Container.
+     *
+     * @param   string         $class      Full class name of the resource.
+     * @param   callable       $factory    Callback to create the class instance. The callback must return instance of the given class.
+     * @param   boolean        $shared     True to create and store a shared instance.
+     * @param   boolean        $protected  True to protect this item from being overwritten. Useful for services.
+     *
+     * @return  $this
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    final public function lazy(string $class, callable $factory, bool $shared = false, bool $protected = false): static
+    {
+        if (PHP_VERSION_ID < 80400) {
+            return $this->set($class, $factory, $shared, $protected);
+        }
+
+        // Create a Lazy Proxy factory
+        $lazyFactory = function () use ($class, $factory) {
```

Comment:
What about using arrow functions here and below?

Reply:
So why not using them in PHP as well?

Reply:
I will keep it as it is